templates/test/ci/prow-machine-pool-ci-version/patches/machine-pool-ci-version.yaml

```diff
@@ -57,6 +57,14 @@ spec:
             done
             systemctl restart kubelet
           fi
+          for CI_CONTAINER in "$${CONTAINERS_TO_TEST[@]}"; do
```

Comment:
this was intentionally removed as we discovered it wasn't needed in https://github.com/kubernetes-sigs/cluster-api/pull/6590#issuecomment-1149463509

Are you observing that it is indeed needed?

cc @jsturtevant @mboersma

Reply:
Just built from main and `kubeadm join` is broken:

```
[2022-06-17 14:38:39] error execution phase preflight: [preflight] Some fatal errors occurred:
[2022-06-17 14:38:39]   [ERROR ImagePull]: failed to pull image registry.k8s.io/kube-apiserver:v1.25.0-alpha.1.29_19c8a2127177b4: output: time="2022-06-17T14:38:34Z" level=fatal msg="pulling image: rpc error: code = NotFound desc = failed to pull and unpack image \"registry.k8s.io/kube-apiserver:v1.25.0-alpha.1.29_19c8a2127177b4\": failed to resolve reference \"registry.k8s.io/kube-apiserver:v1.25.0-alpha.1.29_19c8a2127177b4\": registry.k8s.io/kube-apiserver:v1.25.0-alpha.1.29_19c8a2127177b4: not found"
[2022-06-17 14:38:39] , error: exit status 1
[2022-06-17 14:38:39]   [ERROR ImagePull]: failed to pull image registry.k8s.io/kube-controller-manager:v1.25.0-alpha.1.29_19c8a2127177b4: output: time="2022-06-17T14:38:36Z" level=fatal msg="pulling image: rpc error: code = NotFound desc = failed to pull and unpack image \"registry.k8s.io/kube-controller-manager:v1.25.0-alpha.1.29_19c8a2127177b4\": failed to resolve reference \"registry.k8s.io/kube-controller-manager:v1.25.0-alpha.1.29_19c8a2127177b4\": registry.k8s.io/kube-controller-manager:v1.25.0-alpha.1.29_19c8a2127177b4: not found"
[2022-06-17 14:38:39] , error: exit status 1
[2022-06-17 14:38:39]   [ERROR ImagePull]: failed to pull image registry.k8s.io/kube-scheduler:v1.25.0-alpha.1.29_19c8a2127177b4: output: time="2022-06-17T14:38:37Z" level=fatal msg="pulling image: rpc error: code = NotFound desc = failed to pull and unpack image \"registry.k8s.io/kube-scheduler:v1.25.0-alpha.1.29_19c8a2127177b4\": failed to resolve reference \"registry.k8s.io/kube-scheduler:v1.25.0-alpha.1.29_19c8a2127177b4\": registry.k8s.io/kube-scheduler:v1.25.0-alpha.1.29_19c8a2127177b4: not found"
[2022-06-17 14:38:39] , error: exit status 1
[2022-06-17 14:38:39]   [ERROR ImagePull]: failed to pull image registry.k8s.io/kube-proxy:v1.25.0-alpha.1.29_19c8a2127177b4: output: time="2022-06-17T14:38:39Z" level=fatal msg="pulling image: rpc error: code = NotFound desc = failed to pull and unpack image \"registry.k8s.io/kube-proxy:v1.25.0-alpha.1.29_19c8a2127177b4\": failed to resolve reference \"registry.k8s.io/kube-proxy:v1.25.0-alpha.1.29_19c8a2127177b4\": registry.k8s.io/kube-proxy:v1.25.0-alpha.1.29_19c8a2127177b4: not found"
[2022-06-17 14:38:39] , error: exit status 1
```

So yes I think we need this (or some working) change to restore pulling from latest-style builds.

Reply:
Can you please post the corresponding KCP resource?

Reply:
wait, it is here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/6424bd5bbfeaf42265664eaacb81072babd65727/templates/test/ci/cluster-template-prow-ci-version.yaml#L90

Reply:
confused, the main jobs are passing: https://testgrid.k8s.io/provider-azure-master-signal#capz-conformance

Reply:
Which template is being used with the failure?

Reply:
I think that was the job: https://testgrid.k8s.io/provider-azure-cloud-provider-azure#cloud-provider-azure-slow-capz
But it's also passing again

Reply:
Fail repro requires 3+ control planes (kubeam join is failing but not kubeadm init) + a  "latest" `KUBERNETES_VERSION` input to capz's `ci-entrypoint.sh` script.

None of the standard jobs that run latest bits uses 3 or more control plane nodes.

Reply:
This is what you get when the 2nd control plane node comes online to join (via `/var/log/cloud-init-output.log`):

```
[2022-06-17 16:43:00]   [ERROR ImagePull]: failed to pull image registry.k8s.io/kube-apiserver:v1.25.0-alpha.1.29_19c8a2127177b4: output: time="2022-06-17T16:42:48Z" level=fatal msg="pulling image: rpc error: code = NotFound desc = failed to pull and unpack image \"registry.k8s.io/kube-apiserver:v1.25.0-alpha.1.29_19c8a2127177b4\": failed to resolve reference \"registry.k8s.io/kube-apiserver:v1.25.0-alpha.1.29_19c8a2127177b4\": registry.k8s.io/kube-apiserver:v1.25.0-alpha.1.29_19c8a2127177b4: not found"
[2022-06-17 16:43:00] , error: exit status 1
[2022-06-17 16:43:00]   [ERROR ImagePull]: failed to pull image registry.k8s.io/kube-controller-manager:v1.25.0-alpha.1.29_19c8a2127177b4: output: time="2022-06-17T16:42:49Z" level=fatal msg="pulling image: rpc error: code = NotFound desc = failed to pull and unpack image \"registry.k8s.io/kube-controller-manager:v1.25.0-alpha.1.29_19c8a2127177b4\": failed to resolve reference \"registry.k8s.io/kube-controller-manager:v1.25.0-alpha.1.29_19c8a2127177b4\": registry.k8s.io/kube-controller-manager:v1.25.0-alpha.1.29_19c8a2127177b4: not found"
[2022-06-17 16:43:00] , error: exit status 1
[2022-06-17 16:43:00]   [ERROR ImagePull]: failed to pull image registry.k8s.io/kube-scheduler:v1.25.0-alpha.1.29_19c8a2127177b4: 
o pull and unpack image \"registry.k8s.io/kube-scheduler:v1.25.0-alpha.1.29_19c8a2127177b4\": failed to resolve reference \"registry.k8s.io/kube-scheduler:v1.25.0-alpha.1.29_19c8a2127177b4\": registry.k8s.io/kube-scheduler:v1.25.0-alpha.1.29_19c8a2127177b4: not found"
[2022-06-17 16:43:00] , error: exit status 1
[2022-06-17 16:43:00]   [ERROR ImagePull]: failed to pull image registry.k8s.io/kube-proxy:v1.25.0-alpha.1.29_19c8a2127177b4: output: time="2022-06-17T16:42:52Z" level=fatal msg="pulling image: rpc error: code = NotFound desc = failed to pull and unpack image \"registry.k8s.io/kube-proxy:v1.25.0-alpha.1.29_19c8a2127177b4\": failed to resolve reference \"registry.k8s.io/kube-proxy:v1.25.0-alpha.1.29_19c8a2127177b4\": registry.k8s.io/kube-proxy:v1.25.0-alpha.1.29_19c8a2127177b4: not found"
[2022-06-17 16:43:00] , error: exit status 1
```

Reply:
is the crd for the second control plane node getting `imageRepository` set to `k8s.gcr.io`  in `kind: KubeadmControlPlane` as mentioned in https://github.com/kubernetes-sigs/cluster-api/pull/6590#issuecomment-1149463509?  seems like there might be a bug in the capi reconciler that isn't using the original template?